crates/partition-store/src/snapshots.rs

```diff
@@ -0,0 +1,81 @@
+use std::ops::RangeInclusive;
+use std::path::PathBuf;
+
+use rocksdb::LiveFile;
+use serde::{Deserialize, Serialize};
+use serde_with::base64::Base64;
+use serde_with::{serde_as, DeserializeAs, SerializeAs};
+
+use restate_types::identifiers::{PartitionId, PartitionKey};
+use restate_types::logs::Lsn;
+
+/// Partition snapshot description.
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PartitionSnapshotMetadata {
+    /// Restate cluster name which produced the snapshot.
+    pub cluster_name: String,
+
+    // todo(pavel): the partition table version may be useful for sanity check on restore
+    /// Restate partition id.
+    pub partition_id: PartitionId,
+
+    /// The partition key range that the partition processor which generated this snapshot was
+    /// responsible for, at the time the snapshot was generated.
+    pub key_range: RangeInclusive<PartitionKey>,
+
+    /// The minimum LSN guaranteed to be applied in this snapshot. The actual
+    /// LSN may be >= [minimum_lsn].
+    pub min_applied_lsn: Lsn,
```

Comment:
Perhaps we can get the version from the checkpoint, then create the export, this way we know the exact value rather than the minimum.

Reply:
I was hoping to implement exactly that; unfortunately the checkpointing and export are bundled in the RocksDB API - constructing the Checkpoint[^1] C++ object does nothing, it's just a handle. The alternative is to import the snapshot as a temporary column family, possibly in a separate database, which didn't seem worth it. Can you think of another way?

[1] https://github.com/facebook/rocksdb/blob/0e04ef1a96d75acdcf2cf1802692375d62aea64c/include/rocksdb/utilities/checkpoint.h#L24-L25